promgen/views.py

```diff
@@ -635,6 +635,9 @@ def post(self, request, pk, source):
             logger.info("Importing %s from %s", farm.name, source)
             farm.refresh()
             messages.info(request, "Refreshed hosts")
+            if source != discovery.FARM_DEFAULT:
+                farm.owner = self.request.user
```

Comment:
Wouldn't this reassign the user even if the owner field is already set? 
If the purpose is to ensure owner field to be non-null, then I don't think that modification is needed. Could the conditional perhaps check if the owner field is null?

Reply:
> Wouldn't this reassign the user even if the owner field is already set?

No, because this is inside an higher level `if created:` block. See line [634](https://github.com/line/promgen/pull/599/files#diff-6a13b7546fe1afc70f36c1c1696e55b1d573a559fb49701596187d219943591aR634).
So the code only runs if a new Farm is created after is linked for the first time.

Reply:
Oh, you are right.